lib/classes/CLI.js

```diff
@@ -146,9 +146,26 @@ class CLI {
   }
 
   generateCommandsHelp(commandsArray) {
-    const command = this.serverless.pluginManager.getCommand(commandsArray);
     const commandName = commandsArray.join(' ');
 
+    // Get all the commands using getCommands() with filtered entrypoint
+    // commands and reduce to the required command.
+    const allCommands = this.serverless.pluginManager.getCommands();
+    const command = _.reduce(commandsArray, (currentCmd, cmd) => {
+      if (currentCmd.commands && cmd in currentCmd.commands) {
+        return currentCmd.commands[cmd];
+      }
+      return undefined;
+    }, { commands: allCommands });
```

Comment:
@HyperBrain implemented this based on your idea. Seems to work. Hope this is how you intended it to be :)